crates/flox/src/commands/package.rs

```diff
@@ -277,6 +285,11 @@ impl Publish {
 
         // validate arguments
 
+        let stability = config.override_stability(self.stability).context(indoc! {"
+            Stability is required!
+            Provide using `--stability` or the `stability` config key
+        "})?;
```

Comment:
I don't fully understand the reasoning to make this required, so I'm not sure whether I completely agree with it. Perfecting publish isn't a priority right now though, so I think I'd just ask for a big fat comment explaining the reasons to require it for now.

Also, just to double check I'm not misundertanding, am I correct that the functional stability changes are:
- requiring stability for publish
- setting an env var with stability
- supporting (but not requiring) stability for all development commands

Reply:
Functionally only the publish handling changed from defaulting to "unspecified" to _Error_.

we supported (optional) stabilities in all development commands before (and set a variable): https://github.com/flox/flox/pull/215/files#diff-3bbe0f3376da952edd88493df9a55969a3b12f1e60f8c4bd0625fe19dc6ebc9aL195-L203

The implementation changed only by removing the "Unspecified" variant and wrapping the uses of stability in an `Option`. 

```diff
  enum Stability {
    Stable,
    Staging,
    Unstable,
-   Unspecified,
  }

- if stability != Stability::Unspecified {
+ if let Some(stability) = stability {
  ...
  }
```


> *(the only functional change with this is that you cant opt-out of stabilites via the 
 flag if you already opted-in to stabilities using the config)*